src/supplemental/mqtt/mqtt_msg.c

```diff
@@ -16,6 +16,18 @@ nni_proto_data_init(nni_mqtt_proto_data *proto_data)
 {
 	proto_data->var_header.connect.conn_flags.clean_session = true;
 	proto_data->var_header.connect.keep_alive = 30;
+	proto_data->var_header.connack.properties = NULL;
```

Comment:
It's a union here, the initialize seems improper?

Reply:
still a pointer here

Reply:
I guess we don't need this function, all we need is just `memset` proto_data is zero.

Reply:
you are right